### DISCUSSION TITLE
Is there a way to enable borders on floating windows without enabling SSD?

### DISCUSSION THREAD

**Discussion by charbelnicolas** [labwc/labwc#1549] [Q&A]
Is there a way to enable borders on floating windows without enabling SSD?

I ask this because I use all of my apps maximized without any borders but when a floating window comes up it doesn't have any borders either. Sometimes this makes it difficult to distinguish between the floating window (i.e blender render window) and the app behind it (i.e blender itself).

Thank you, any help is appreciated!

**Comment by dmitry-azaraev** [comment]
@charbelnicolas when SSD enabled, maximized apps already have no border.

labwc implements 3-state SSD:, no ssd, only borders and borders+title. So you generally should be able somehow configure it (not sure), to use borders only (which is supressed when maximized anyway). However, as i understand there is app preference to use SSD or no.

I guess tweaking window rules with proper actions might help get desired result.

**Comment by dmitry-azaraev** [comment]
E.g. idea is add window rule which will trigger action ToggleDecorations to fit your needs. See https://labwc.github.io/labwc-actions.5.html .

**Reply by charbelnicolas** [reply]
Thank you, I'll check it out. I have already dIsabled SSD for all windows with rules but I do want the border for floating windows and no SSD.

**Comment by charbelnicolas** [comment]
I was able to achieve it with the following rules:

```
<windowRule identifier="*" fixedPosition="yes">
	<action name="ToggleDecorations" />
	<action name="ResizeTo" width="1918" height="1078" />
	<action name="FitToOutput" />
</windowRule>
```

It seems kinda hacky though... I wish there was a cleaner way...

**Reply by Consolatis** [reply]
We indeed need some `SetDecoration` action which allows to target one of the 3 states specifically (e.g. all, border, none). The `ToggleDecorations` action is acting based on the current state of a window so is not deterministic at all. This is fine for the usual use-case of window-menus or keybinds but pretty annoying for window rules.

**Reply by charbelnicolas** [reply]
Hey @Consolatis I'm glad you've brought this up! I was about to make a detailed issue about this but it seems you're pretty aware of it.

The solution with window rules doesn't really work since you get a gap in between your window and the current output edges unless you use the ResizeTo action.

I was thinking about having an option in the  `<decoration>server</decoration>` to set it to `none` or `borders`.

The following window rule causes issues with splash screens and floating windows like file choosers and certain dialogs since it resizes those windows too.

```
<windowRule identifier="*" fixedPosition="yes">
	<action name="ToggleDecorations" />
	<action name="ResizeTo" width="1918" height="1078" />
	<action name="FitToOutput" />
</windowRule>
``` 

Like you said, I wish there was just a setting to enable borders only and be done with it. Let's be honest if you're using labwc you are mostly doing everything with keybinds (at least I am), that means closing, maximizing and minimizing windows and all those basic things. Server side decorations take valuable vertical space on a 1080p monitor which is not desirable.

Borders are important for non-maximized or non-fullscreen windows because they allow you to distinguish one window from another when the windows have very similar background colors (i.e using Blender with a render window on top).

I'd be stoked to have this in labwc, it's one of the few things that have caused some annoyances while using labwc coming from awesomewm.

I'd really appreciate if this was taken into consideration.

Thank you!

**Comment by charbelnicolas** [comment]
And those rules won't work for xdg-desktop-portal-gtk, I had to add a specific rule for it:

```
<windowRule identifier="xdg-desktop-portal-gtk" fixedPosition="yes">
	<action name="ToggleDecorations" />
	<action name="Maximize" />
</windowRule>
```
